src/listbox/Listbox.ts

```diff
@@ -0,0 +1,264 @@
+			case Keys.Enter:
+			case Keys.Space:
+				event.preventDefault();
+				if (!getOptionDisabled(activeItem, activeIndex)) {
+					onOptionSelect && onOptionSelect(activeItem, activeIndex, key);
+				}
+				break;
+			case Keys.Down:
+				event.preventDefault();
+				newIndex = (activeIndex + 1) % optionData.length;
+				onActiveIndexChange && onActiveIndexChange(newIndex, key);
+				break;
+			case Keys.Up:
+				event.preventDefault();
+				newIndex = (activeIndex - 1 + optionData.length) % optionData.length;
+				onActiveIndexChange && onActiveIndexChange(newIndex, key);
+				break;
+			case Keys.Home:
+				onActiveIndexChange && onActiveIndexChange(0, key);
+				break;
+			case Keys.End:
+				onActiveIndexChange && onActiveIndexChange(optionData.length - 1, key);
+				break;
+		}
+	}
+
+	protected animateScroll(element: HTMLElement, scrollValue: number) {
+		element.scrollTop = scrollValue;
+	}
+
+	@diffProperty('activeIndex', auto)
+	protected calculateScroll(previousProperties: ListboxProperties, { activeIndex = 0 }: ListboxProperties) {
+		const scrollOffset = this.meta(Dimensions).get('root').scroll.top;
+		const menuHeight = this.meta(Dimensions).get('root').offset.height;
+		const optionOffset = this.meta(Dimensions).get(this._getOptionId(activeIndex)).offset;
+
+		if (optionOffset.top - scrollOffset < 0) {
+			this._scroll = optionOffset.top;
+			this.invalidate();
+		}
+
+		else if ((optionOffset.top + optionOffset.height) > (scrollOffset + menuHeight)) {
+			this._scroll = optionOffset.top + optionOffset.height - menuHeight;
+			this.invalidate();
+		}
+	}
+
+	protected getModifierClasses() {
+		const { visualFocus } = this.properties;
+		return [
+			visualFocus ? css.focused : null
+		];
+	}
+
+	protected getOptionClasses(active: boolean, disabled: boolean, selected: boolean) {
+		return [
+			css.option,
+			active ? css.activeOption : null,
+			disabled ? css.disabledOption : null,
+			selected ? css.selectedOption : null
+		];
+	}
+
+	protected onElementCreated(element: HTMLElement, key: string) {
+		if (key === 'root' && typeof this._scroll === 'number') {
+			this.animateScroll(element, this._scroll);
+			this._scroll = undefined;
+		}
+	}
+
+	protected onElementUpdated(element: HTMLElement, key: string) {
+		if (key === 'root' && typeof this._scroll === 'number') {
+			this.animateScroll(element, this._scroll);
+			this._scroll = undefined;
+		}
+	}
+
+	protected renderOptionLabel(option: any, index: number): DNode {
+		const { getOptionLabel } = this.properties;
+		return getOptionLabel ? getOptionLabel(option, index) : `${option}`;
+	}
+
+	protected renderOption(option: any, index: number): DNode {
+		const {
+			activeIndex = 0,
+			key = '',
+			getOptionDisabled = (option: any, index: number) => false,
+			getOptionSelected = (option: any, index: number) => false,
+			theme,
+			onActiveIndexChange,
+			onOptionSelect
+		} = this.properties;
+
+		const disabled = getOptionDisabled(option, index);
```

Comment:
It looks like you call this function a few different places, and also define it in each of those places. Could you pull this out into a static function somewhere? Or consider using a default property.

Reply:
Never mind, default properties haven't landed.